libs/wingsdk/src/shared-aws/bucket.inflight.ts

```diff
@@ -263,17 +263,29 @@ export class BucketClient implements IBucketClient {
     );
   }
 
-  /**
-   * Returns a signed url to the given file. This URL can be used by anyone to
-   * access the file until the link expires (defaults to 24 hours).
+  
+ /**
+   * Returns a signed url to the given file.
+   * @Throws if object does not exist.
+   * @inflight
    * @param key The key to reach
-   * @param duration Time until expires
+*    @param duration Time until expires
    */
-  public async signed_url(key: string, duration?: Duration): Promise<string> {
-    // for signed_url take a look here: https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/s3-example-creating-buckets.html#s3-create-presigendurl-get
-    throw new Error(
-      `signed_url is not implemented yet (key=${key}, duration=${duration})`
-    );
+
+
+  public async signedUrl(key: string, duration?: Duration): Promise<string> {
+    if (!(await this.exists(key))) {
+      throw new Error(
+        `Cannot provide signed url for a non-existent key (key=${key})`
+      );
+    }
+    const expiryTimeInSeconds:number= duration?.seconds|| 86400;
```

Comment:
Why did you choose this default?

Reply:
@revitalbarletz If the duration is not provided , Shouldn't it be defaulted to 24 hours(86400 seconds)?

This is the detail given in the spec
@revitalbarletz Let me know what to change here?

Reply:
That's okay! I was just wondering if it comes from the spec.